parsec/core/mountpoint/manager.py

```diff
@@ -112,19 +119,14 @@ def get_path_in_mountpoint(
         if timestamp is None:
             workspace = self._get_workspace(workspace_id)
         else:
-            workspace = await self._get_workspace_timestamped(workspace_id, timestamp)
+            try:
+                workspace = self._get_workspace_timestamped(workspace_id, timestamp)
```

Comment:
Hmm, `workspace` is useless here right? This block is going to raise in any case right?
```
        if (workspace_id, timestamp) in self._mountpoint_tasks:
            raise MountpointAlreadyMounted(f"Workspace `{workspace_id}` already mounted.")
```
So why not structure it like so:
```python
    async def mount_workspace(self, workspace_id: EntryID, timestamp: Pendulum = None) -> PurePath:
        if (workspace_id, timestamp) in self._mountpoint_tasks:
            raise MountpointAlreadyMounted(f"Workspace `{workspace_id}` already mounted.")
        if timestamp is not None:
            return await self.remount_workspace_new_timestamp(workspace_id, None, timestamp)
        workspace = self._get_workspace(workspace_id)
        runner_task = await self._mount_workspace_helper(workspace)
        return runner_task.value
```